clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4493,6 +4493,13 @@ static void RenderDiagnosticsOptions(const Driver &D, const ArgList &Args,
 
   Args.addOptOutFlag(CmdArgs, options::OPT_fspell_checking,
                      options::OPT_fno_spell_checking);
+
+  if (const Arg *A =
+          Args.getLastArg(options::OPT_warning_suppression_mappings_EQ)) {
+    if (!D.getVFS().exists(A->getValue()))
+      D.Diag(clang::diag::err_drv_no_such_file) << A->getValue();
```

Comment:
I'm not super keen on this because it's a TOCTOU issue. We should issue this diagnostic if we're not able to open the file at the time we need to open it. That also covers the case where the file exists but doesn't have read permissions.